site/content/docs/howto/configure-supervisor-with-gitlab.md

```diff
@@ -0,0 +1,83 @@
+---
+title: Configure the Pinniped Supervisor to use Gitlab as an OIDC Provider
+description: Set up the Pinniped Supervisor to use Gitlab login.
+cascade:
+  layout: docs
+menu:
+  docs:
+    name: Configure Supervisor With Gitlab
+    weight: 35
+    parent: howtos
+---
+The Supervisor is an [OpenID Connect (OIDC)](https://openid.net/connect/) issuer that supports connecting a single "upstream" OIDC identity provider to many "downstream" cluster clients.
+
+This guide shows you how to configure the supervisor so that users can authenticate to their Kubernetes
+cluster using their Gitlab credentials.
+## Prerequisites
+
+This how-to guide assumes that you have already installed the Pinniped Supervisor with working ingress,
+and that you have configured a `FederationDomain` to issue tokens for your downstream clusters, as
+described [here](https://pinniped.dev/docs/howto/configure-supervisor/).
+
+## Configuring your Gitlab Application
+1. In Gitlab, navigate to User Settings > Applications
+1. Create a new application:
+   1. Enter the name of your application.
+   1. Enter the redirect URI. This is the `issuer` you configured in your `FederationDomain` appended with `/callback`.
+   1. Check the box saying that the application is Confidential.
+   1. Select scope `openid`. Optionally select `profile` and `email`.
+   1. Save the application and make note of the Application ID and Secret.
+
+## Configuring the Supervisor cluster
+Create an [`OIDCIdentityProvider`](https://github.com/vmware-tanzu/pinniped/blob/main/generated/1.20/README.adoc#oidcidentityprovider) in the same namespace as the Supervisor.
+```yaml
+apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+kind: OIDCIdentityProvider
+metadata:
+  name: my-oidc-provider
+spec:
+  # The upstream issuer name.
+  # This should be something like https://gitlab.com or https://gitlab.your-company-name.example.com.
+  issuer: "<gitlab-url>"
+  # Optionally specify the CA bundle for the GitLab server as base64 encoded PEM data.
+  tls:
+    certificateAuthorityData: "<gitlab-ca-bundle>"
+  authorizationConfig:
+    # Any scopes other than "openid" that you selected when creating your GitLab application.
+    additionalScopes: [ email, profile ]
+  # See here for a list of available claims: https://docs.gitlab.com/ee/integration/openid_connect_provider.html#shared-information
+  claims:
+    # The name of the claim in your GitLab token that will be mapped to the "username" claim in downstream
+    # tokens minted by the Supervisor.
+    # For example, "email" or "sub".
+    username: "<username-claim>"
+    # The name of the claim in your GitLab token that represents the groups that the user belongs to.
```

Comment:
does gitlab include groups in their tokens? if users can only get their groups from the userinfo endpoint, then is it worth mentioning that here?

Reply:
Only in the userinfo endpoint.

Reply:
Do you think it's useful just as a note? I don't think it's actionable.

Reply:
agreed that it isn't actionable. when i read "your GitLab token" it made me think that the ID token that gitlab sends back would have the groups in it. but maybe that isn't even helpful. up to you.